test/fixtures/dynamic-element.js

```diff
@@ -0,0 +1,14 @@
+export default ({ level = 1 }) => {
+  const Element = `h${level}`
+
+  return (
+    <Element className="root">
+      <p>dynamic element</p>
+      <style jsx>{`
+        .root {
+          background: red;
+        }
+      `}</style>
+    </Element>
+  )
+}
```

Comment:
awesome can you add a few more tests cases? i.e. a few more combinations to where `Element` is defined outside of the component, and maybe when it is lowercase too (any edge case you can think of). Also can you add a class component too?

Reply:
Maybe Im missing something, but I dont think it will work if Element is defined outside of the component. ( Thats why I added the styled-jsx option )
I´ll add the testcases

Reply:
correct, it won't work (it won't add the className) and we want a snapshot of that! ;)

Reply:
👍